huaweicloud/services/hss/resource_huaweicloud_hss_quota.go

```diff
@@ -0,0 +1,436 @@
+	}
+
+	quotas, err := GetQuotaById(client, id, epsId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(quotas) < 1 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "HSS quota")
+	}
+
+	quota := quotas[0]
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("version", utils.PathSearch("version", quota, nil)),
+		d.Set("status", utils.PathSearch("quota_status", quota, nil)),
+		d.Set("used_status", utils.PathSearch("used_status", quota, nil)),
+		d.Set("host_id", utils.PathSearch("host_id", quota, nil)),
+		d.Set("host_name", utils.PathSearch("host_name", quota, nil)),
+		d.Set("charging_mode", convertChargingMode(utils.String(utils.PathSearch("charging_mode", quota, "").(string)))),
+		d.Set("expire_time", utils.PathSearch("expire_time", quota, nil)),
+		d.Set("shared_quota", utils.PathSearch("shared_quota", quota, nil)),
+		d.Set("is_trial_quota", utils.PathSearch("is_trial_quota", quota, nil)),
+		d.Set("enterprise_project_id", utils.PathSearch("enterprise_project_id", quota, nil)),
+		d.Set("enterprise_project_name", utils.PathSearch("enterprise_project_name", quota, nil)),
+		d.Set("tags", utils.FlattenTagsToMap(utils.PathSearch("tags", quota, nil))),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceQuotaUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		id      = d.Id()
+		product = "hss"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating HSS client: %s", err)
+	}
+
+	if d.HasChange("auto_renew") {
+		bssClient, err := cfg.BssV2Client(region)
+		if err != nil {
+			return diag.Errorf("error creating BSS V2 client: %s", err)
+		}
+
+		if err = common.UpdateAutoRenew(bssClient, d.Get("auto_renew").(string), id); err != nil {
+			return diag.Errorf("error updating the auto_renew of the HSS quota (%s): %s", id, err)
+		}
+	}
+
+	if d.HasChange("tags") {
+		oRaw, nRaw := d.GetChange("tags")
+		oMap := oRaw.(map[string]interface{})
+		nMap := nRaw.(map[string]interface{})
+
+		if len(oMap) > 0 {
+			oldKeys := getOldTagKeys(oMap)
+			if err := utils.DeleteResourceTagsWithKeys(client, oldKeys, "hss", id); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+		if len(nMap) > 0 {
+			if err := CreateQuotaTags(client, id, nMap); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	if d.HasChange("enterprise_project_id") {
+		migrateOpts := enterpriseprojects.MigrateResourceOpts{
+			ResourceId:   id,
+			ResourceType: "hss",
+			RegionId:     region,
+			ProjectId:    client.ProjectID,
+		}
+		if err := common.MigrateEnterpriseProject(ctx, cfg, d, migrateOpts); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return resourceQuotaRead(ctx, d, meta)
+}
+
+func CreateQuotaTags(client *golangsdk.ServiceClient, id string, tagsMap map[string]interface{}) error {
```

Comment:
Please name methods in lowercase letters.

Reply:
done.